### PR TITLE
AIRFLOW-105 Implement SqoopHook

### PR DESCRIPTION
**Pull request by bolkedebruin**
Dear Airflow Maintainers,

Please accept this PR that addresses the following issues:
- https://issues.apache.org/jira/browse/AIRFLOW-105

This implements an initial Sqoop 1 Hook.

### CODE REVIEW

File: airflow/contrib/hooks/sqoop_hook.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+"""
+from airflow.hooks.base_hook import BaseHook
+from airflow.exceptions import AirflowException
+
+import logging
+import subprocess
+
+log = logging.getLogger(__name__)
+
+class SqoopHook(BaseHook):
```
Comment (mistercrunch) [inline airflow/contrib/hooks/sqoop_hook.py]: docstring please! It's important to document the extra params well.

**Comment by mistercrunch** [comment]
- entry in `docs/code.rst`

File: airflow/contrib/hooks/sqoop_hook.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        cmd = self._import_cmd(target_dir, append, type,
+                               num_mappers, split_by)
+        cmd += ["--table", table]
+        if columns:
+            cmd += ["--columns", columns]
+        if where:
+            cmd += ["--where", where]
+
+        logging.debug(cmd)
+        print(cmd)
```
Comment (mistercrunch) [inline airflow/contrib/hooks/sqoop_hook.py]: NIT: remove line

**Comment by bolkedebruin** [comment]
@mistercrunch any remaining issues for you?

**Comment by bolkedebruin** [comment]
Is superseded.

**Pull request closed without merging**